substratevm/src/com.oracle.graal.pointsto/src/com/oracle/graal/pointsto/api/SharedHostVM.java

```diff
@@ -0,0 +1,155 @@
+    private final OptionValues options;
+    protected final ClassLoader classLoader;
+
+    protected final List<BiConsumer<AnalysisMethod, StructuredGraph>> methodAfterParsingHooks = new CopyOnWriteArrayList<>();
+
+    public void addMethodAfterParsingHook(BiConsumer<AnalysisMethod, StructuredGraph> methodAfterParsingHook) {
+        methodAfterParsingHooks.add(methodAfterParsingHook);
+    }
+
+    protected SharedHostVM(OptionValues options, ClassLoader classLoader) {
+        this.options = options;
+        this.classLoader = classLoader;
+        this.classReachabilityListeners = new ArrayList<>();
+    }
+
+    @Override
+    public OptionValues options() {
+        return options;
+    }
+
+    @Override
+    public boolean isRelocatedPointer(Object originalObject) {
+        return false;
+    }
+
+    @Override
+    public void clearInThread() {
+
+    }
+
+    @Override
+    public void installInThread(Object vmConfig) {
+        Thread.currentThread().setContextClassLoader(classLoader);
+    }
+
+    @Override
+    public Object getConfiguration() {
+        return null;
+    }
+
+    @Override
+    public void checkForbidden(AnalysisType type, AnalysisType.UsageKind kind) {
+    }
+
+    @Override
+    public void registerType(AnalysisType newValue) {
+
+    }
+
+    @Override
+    public Optional<AnalysisMethod> handleForeignCall(ForeignCallDescriptor foreignCallDescriptor, ForeignCallsProvider foreignCallsProvider) {
+        return Optional.empty();
+    }
+
+    @Override
+    public GraphBuilderPhase.Instance createGraphBuilderPhase(HostedProviders providers, GraphBuilderConfiguration graphBuilderConfig, OptimisticOptimizations optimisticOpts,
+                    IntrinsicContext initialIntrinsicContext) {
+        return null;
+    }
+
+    @Override
+    public String inspectServerContentPath() {
+        return PointstoOptions.InspectServerContentPath.getValue(options);
+    }
+
+    @Override
+    public void warn(String message) {
+        System.err.println("warning: " + message);
+    }
+
+    public void registerClassReachabilityListener(BiConsumer<Feature.DuringAnalysisAccess, Class<?>> listener) {
+        classReachabilityListeners.add(listener);
+    }
+
+    public void notifyClassReachabilityListener(AnalysisUniverse universe, Feature.DuringAnalysisAccess access) {
+        for (AnalysisType type : universe.getTypes()) {
+            if (type.isReachable() && !type.getReachabilityListenerNotified()) {
+                type.setReachabilityListenerNotified(true);
+
+                for (BiConsumer<Feature.DuringAnalysisAccess, Class<?>> listener : classReachabilityListeners) {
+                    listener.accept(access, type.getJavaClass());
+                }
+            }
+        }
+    }
+
+    @Override
+    public void checkType(ResolvedJavaType type, AnalysisUniverse universe) {
+
+    }
+
+    @Override
+    public void methodAfterParsingHook(BigBang bb, AnalysisMethod method, StructuredGraph graph) {
```

Comment:
You can remove this, it is implemented by both subclasses.

Reply:
Fixed